File: Solutions/3016- Minimum Number of Pushes to Type Word II/Solution.py
class Solution:
    def minimumPushes(self, word: str) -> int:
        my_dict = Counter(word)

        if len(my_dict.keys()) < 9:
            return len(word)

        answer = 0
        sorted_keys = [key for key, value in sorted(my_dict.items(), key=lambda item: item[1], reverse=True)]
        for i in range(0,len(sorted_keys)):
            answer += my_dict[sorted_keys[i]] * (1 + (i // 8))
        
        return answer

File: Solutions/3016- Minimum Number of Pushes to Type Word II/Solution.java
import java.util.*;

class Solution {
    public int minimumPushes(String word) {
        int[] frequency = new int[26]; 
        
        for (char c : word.toCharArray()) {
            frequency[c - 'a']++;
        }

        List<Integer> counts = new ArrayList<>();
        for (int count : frequency) {
            if (count > 0) {
                counts.add(count);
            }
        }

        Collections.sort(counts, Collections.reverseOrder());
        int answer = 0;
        
        for (int i = 0; i < counts.size(); i++) {
            answer += counts.get(i) * (1 + (i / 8));
        }

        return answer;
    }
}
